### PR TITLE
Update validatePaths type

### PR DESCRIPTION
**Pull request by eviemayy**
By adding type `Array<RegExp>` in addition to `Array<string>`, the `validatePaths` parameter can now be used to validate a more specific set of paths.

For example:
Using the regex pattern: `/pets(?!/123)` would allow for paths that start with `/pets` but would _not_ allow the full path `pets/123`.

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -109,7 +109,7 @@ export function validateConfig(cfg: Partial<Config>): Config {
         },
         enableTypes: ['form']
       })
-    },
```
Comment (supertong) [inline src/config.ts:112]: Why is this required now?

File: src/config.ts [hunk 2]
```diff
@@ -109,7 +109,7 @@ export function validateConfig(cfg: Partial<Config>): Config {
...
+    } as any,
```
Comment (supertong) [inline src/config.ts:112]: Why is this `as any` required now?

File: src/index.ts [hunk 3]
```diff
@@ -151,3 +151,18 @@ async function compileOas(config: Config) {
...
+    if (isRegexPattern(path)) {
+      return (path as RegExp).test(ctx.path);
+    } else {
+      return ctx.path.startsWith(path as string);
+    }
+  });
+  return dontValidate;
+}
+
+function isRegexPattern(path: string | RegExp) {
```
Comment (supertong) [inline src/index.ts:166]: Can you make this a type guard function so that we don't have to cast to `RegExp` in the above?

File: src/index.ts [hunk 4]
```diff
@@ -151,3 +151,18 @@ async function compileOas(config: Config) {
...
+  return typeof path == 'object' ? true : false;
```
Comment (supertong) [inline src/index.ts:167]: Can we do `path instanceof RegExp` instead?

**Comment by supertong** [comment]
Hey @eviemayy , thanks for contribution! I have left a few comment if you could take a look.
Cheers

File: src/config.ts (hunk 2 above)
Comment (eviemayy) [reply src/config.ts:112]: woops, it's not needed, I've removed this

File: src/index.ts (hunk 4 above)
Comment (eviemayy) [reply src/index.ts]: Good point! I've removed the `isRegexPattern` function altogether and just added `path instanceof RegExp` so that we don't have to type cast in line 158
